Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUBindGroupLayoutImpl.cpp

```diff
@@ -33,10 +33,12 @@
 
 namespace PAL::WebGPU {
 
-BindGroupLayoutImpl::BindGroupLayoutImpl(WGPUBindGroupLayout bindGroupLayout, ConvertToBackingContext& convertToBackingContext)
+BindGroupLayoutImpl::BindGroupLayoutImpl(WGPUBindGroupLayout bindGroupLayout, ConvertToBackingContext& convertToBackingContext, Ownership ownership)
     : m_backing(bindGroupLayout)
     , m_convertToBackingContext(convertToBackingContext)
 {
+    if (ownership == Ownership::Wrap)
+        wgpuBindGroupLayoutRetain(bindGroupLayout);
```

Comment:
There's an alternative way of doing this: instead of retaining the +0 bind group, we can retain the wrapper of the pipeline object which owns the +0 bind group. Maybe that's better than this approach? It would allow us to not expose more WebGPU.framework API additions, which is always good.

Reply:
Mike and I both agree that this approach is better, but requires more thought about how to make this work with the C++ type system. Marking as draft for now.